engine/table/src/main/java/io/deephaven/engine/table/impl/sources/UnionSourceManager.java

```diff
@@ -209,6 +211,20 @@ private final class ConstituentListenerRecorder extends LinkedListenerRecorder {
             setMergedListener(mergedListener);
         }
 
+        @Override
+        protected void onFailureInternal(@NotNull final Throwable originalException, @Nullable final Entry sourceEntry) {
+            // We will just record the error here for now.  If the table was removed, then we don't actually care about it
+            // but if the error is real, and the table is not removed, it will be propagated in processExisting()
+            this.setNotificationStep(getUpdateGraph().clock().currentStep());
+            this.error = originalException;
+            mergedListener.notifyChanges();
+        }
+
+        @Override
+        public boolean recordedVariablesAreValid() {
+            return error == null && super.recordedVariablesAreValid();
```

Comment:
Don't you need to fix this check in MergedListener to use the recordedVariablesAreValid instead?

                for (ListenerRecorder recorder : recorders) {
                    if (recorder.getNotificationStep() == currentStep) {
                        added += recorder.getAdded().size();
                        removed += recorder.getRemoved().size();
                        modified += recorder.getModified().size();
                        shifted += recorder.getShifted().getEffectiveSize();
                    }
                }

Reply:
No,  the recorder will return empty everytthing when recorded values are not valid.  That's why I did it that way

Reply:
The source table may not double notify.  In the case that the source table of the ListenerRecorder notifies us of an error, we should correctly set the notification step and record the error - the listener did have a notification on that step.  Given that there is an error there will never again be a valid notification from that source table.

The update in this case is going to be null; and if we ask the question "did this get updated on the current step" and then fetch and dereference the null update as is done on line 319-326 of MergedListener we have an error.  The change to the recordedVariablesAreValid (which is just asking about the step) means that we get the appropriate empty update in this section of the MergedListener.